search/search_quality/assessment_tool/sample_view.cpp

```diff
@@ -75,6 +72,10 @@ void SampleView::ShowResults(search::Results::Iter begin, search::Results::Iter
 void SampleView::SetResultRelevances(std::vector<Relevance> const & relevances)
 {
   CHECK_EQUAL(relevances.size(), m_results->Size(), ());
+
+  m_edits.ResetRelevances(relevances);
   for (size_t i = 0; i < relevances.size(); ++i)
-    m_results->Get(i).SetRelevance(relevances[i]);
+    m_results->Get(i).EnableEditing(Edits::RelevanceEditor(m_edits, i));
 }
+
+void SampleView::OnUpdate() { emit EditStateUpdated(m_edits.HasChanges()); }
```

Comment:
No, because it overrides Delegate::OnUpdate(). And I don't see why do we need to make Delegate::OnUpdate() constant.